### PR TITLE
fix(scaffolder): update annotator action readme

### PR DESCRIPTION
**Pull request by debsmita1**
**Resolves:**

Addresses the comments on the PR  https://github.com/janus-idp/backstage-plugins/pull/1567

### CODE REVIEW

**Review by Zaperex — COMMENTED** [review]
Should we add more explicit descriptions of what the [`timestamping` and `scaffolded-from` custom actions](https://github.com/janus-idp/backstage-plugins/pull/1638/files#diff-42e97cdf66aae5d9f993fed9fe4770384baecb95d37d1da8ea6bbf84bd71ebabR147-R152) do/modify and potentially provide more information on how they're used. 

Ex: Referring them to the [example directory](https://github.com/janus-idp/backstage-plugins/tree/main/plugins/scaffolder-annotator-action/examples) for examples on how the provided custom actions are used. Telling them that the `timestamping` action adds a `backstage.io/createdAt` annotation with the creation time and the `scaffolded-from` action adding a `scaffoldedFrom` spec with the entityRef of the template that ran the action.

Also I just noticed that the [generic annotator custom action](https://github.com/janus-idp/backstage-plugins/blob/main/plugins/scaffolder-annotator-action/src/actions/annotator/annotator.ts) was [not exported for the new plugin installation](https://github.com/janus-idp/backstage-plugins/blob/b89ca1deff93a088a6a101091350204c3ee0a117/plugins/scaffolder-annotator-action/src/module.ts#L20-L23). I think we should include that since we talked about the capabilities of that generic action in this documentation (potentially add an example where the generic action is used?)

**Review by Zaperex — CHANGES_REQUESTED** [review]

**Comment by nickboldt** [comment]
> @debsmita1 I was just wondering, is the [`getCurrentTimestamp`](https://github.com/janus-idp/backstage-plugins/blob/1b7e4c139589100134361638de907a6f8059098e/plugins/scaffolder-annotator-action/src/utils/getCurrentTimestamp.ts#L4-L5) utility function necessary?
> 
> You should be able to achieve the same thing by using the following instead?
> 
> ```
> const time = `${dateObj.toLocaleDateString()}, ${dateObj.toLocaleTimeString('en-US')}`
> ```
> 
> This is mainly to address the fact that the current tests for that utility function being brittle and compares the output of toLocaleDateString (which can be formatted differently depending on the platform it's run) with a hard-coded date.

for context:

> with my change: [https://productionresultssa2.blob.core.windows.net/actions-results/37472355-c409-468d-81af-fe2176f[…]&sr=b&st=2024-05-28T22%3A47%3A33Z&sv=2021-12-02](https://productionresultssa2.blob.core.windows.net/actions-results/37472355-c409-468d-81af-fe2176f3ff2f/workflow-job-run-cee48acf-97aa-50d9-a49b-7a4d2550230d/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-05-28T22%3A57%3A38Z&sig=pG6TVf3Bl8QtLZk4yb%2FNT5sbDW3M9p%2Bdu7nZZihy0ew%3D&sp=r&spr=https&sr=b&st=2024-05-28T22%3A47%3A33Z&sv=2021-12-02)

```
2024-05-28T21:52:34.6648360Z     Expected: "2021-05-12, 07:03:18 AM"
2024-05-28T21:52:34.6648891Z     Received: "5/12/2021, 07:03:18 AM"
```
but without it I had the opposite failure locally

**Review by Zaperex — CHANGES_REQUESTED** [review]

**Comment by Zaperex** [comment]
A `yarn prettier:fix` needs to be run as well

**Comment by debsmita1** [comment]
> @debsmita1 I was just wondering, is the [`getCurrentTimestamp`](https://github.com/janus-idp/backstage-plugins/blob/1b7e4c139589100134361638de907a6f8059098e/plugins/scaffolder-annotator-action/src/utils/getCurrentTimestamp.ts#L4-L5) utility function necessary?
> 
> You should be able to achieve the same thing by using the following instead?
> 
> ```
> const time = `${dateObj.toLocaleDateString()}, ${dateObj.toLocaleTimeString('en-US')}`
> ```
> 
> This is mainly to address the fact that the current tests for that utility function are brittle and compares the output of `toLocaleDateString` (which can be formatted differently depending on the platform it's run on) with a hard-coded date.

Thank you for catching this, I should have converted the expected date string to localeDateString. Strangely, the CI didn't catch this earlier

**Review by Zaperex — CHANGES_REQUESTED** [review]
I was just wondering if we even need this package since it's just calling the built in Date functions. This function is only used here: https://github.com/janus-idp/backstage-plugins/blob/4e36115e6abfaba172b9f39fa5902bc92ed45bf4/plugins/scaffolder-annotator-action/src/actions/customActions/createTimestampAction.ts#L9C48-L9C67

So would it make sense to replace that function call with an in-line `${dateObj.toLocaleDateString()}, ${dateObj.toLocaleTimeString('en-US')}` and remove the `getCurrentTimestamp` util package along with the test for it?

File: plugins/scaffolder-annotator-action/src/utils/getCurrentTimestamp.ts [hunk 1]
```diff
@@ -1,16 +1,4 @@
-const zeroPad = (d: number) =>
-  d < 10 ? '0'.concat(d.toString()) : d.toString();
-
 export const getCurrentTimestamp = (date?: Date) => {
```
[the remark this answers is not in the document]
Comment (debsmita1) [reply plugins/scaffolder-annotator-action/src/utils/getCurrentTimestamp.ts:1]: Including the timestamp directly in the `createTimestampAction` would mean initializing the dateObj with Date.now within the action. This approach generates the timestamp when the action is added to the instance, whereas we want the timestamp to be added when the annotation is added. Therefore, I still believe it is crucial to have a utility to generate the timestamp.

[the remark this answers is not in the document]
Comment (Zaperex) [reply plugins/scaffolder-annotator-action/src/utils/getCurrentTimestamp.ts:1]: Ah ok that makes sense. If so, these changes LGTM

**Review by Zaperex — APPROVED** [review]
/lgtm

**Pull request merged**